config/config-api/src/com/thoughtworks/go/domain/PipelineGroups.java

```diff
@@ -158,18 +158,17 @@ public void addError(String fieldName, String message) {
         configErrors.add(fieldName, message);
     }
 
-    public HashSet<MaterialConfig> getAllUniquePostCommitSchedulableMaterials() {
-        HashSet<MaterialConfig> materialConfigs = new HashSet<MaterialConfig>();
-        Set<Map> uniqueMaterials = new HashSet<Map>();
+    public Set<MaterialConfig> getAllUniquePostCommitSchedulableMaterials() {
+        Set<MaterialConfig> materialConfigs = new HashSet<MaterialConfig>();
+        Set<String> uniqueMaterials = new HashSet<String>();
         for (PipelineConfigs pipelineConfigs : this) {
             for (PipelineConfig pipelineConfig : pipelineConfigs) {
                 for (MaterialConfig materialConfig : pipelineConfig.materialConfigs()) {
-                    if (!uniqueMaterials.contains(materialConfig.getSqlCriteria()) && materialConfig instanceof ScmMaterialConfig) {
-                        if (materialConfig.isAutoUpdate()) {
-                            continue;
-                        }
+                    if ((materialConfig instanceof ScmMaterialConfig || materialConfig instanceof PluggableSCMMaterialConfig)
+                            && !materialConfig.isAutoUpdate()
+                            && !uniqueMaterials.contains(materialConfig.getFingerprint())) {
                         materialConfigs.add(materialConfig);
-                        uniqueMaterials.add(materialConfig.getSqlCriteria());
+                        uniqueMaterials.add(materialConfig.getFingerprint());
```

Comment:
Can you make a couple of small changes?

1. PluggableSCMMaterialConfig#getFingerprint() is not locally cached, as with other materials.

2. materialConfig.getFingerprint() can be null, especially in the case of pluggable SCM materials. How is this handled? My guess is that this will fail, since it will try to add null into the list. If you think it cannot be null, why is PluggableSCMMaterialConfig#getFingerprint() returning null?